### PR TITLE
feat(slider): Add 2nd slider handle for range

### PR DESCRIPTION
**Pull request by zvonimirfras**
closes #935 #916

### CODE REVIEW

File: src/slider/slider.component.ts [hunk 1]
```diff
@@ -179,6 +191,44 @@ export class Slider implements AfterViewInit, ControlValueAccessor {
 	get value() {
 		return this._value;
 	}
+
+	/** Set the initial value. Available for two way binding */
+	@Input() set value2(v) {
```
Comment (cal-smith) [inline src/slider/slider.component.ts:196]: We should be able to use an array to represent the high and low/left and right values ... That way we aren't specifically coding in `value1` and `value2` ... In theory we should be able to support `N` sliders, but we can start with just a tuple.

So `value` should have the type `number | number[]` ... As long as we eimt a new array each time `ngModel` should work fine with this setup. I would expect that we swap everything internally to use arrays, and just dump the contents of `this.value[0]`. `isRange` can probably just be a check for `this.value.length > 1` 🤔

**Review by cal-smith — CHANGES_REQUESTED** [review]
Similar comments to before ... I'd rather not get stuck with `value` and `value2` if possible 🤔

File: src/slider/slider.component.ts (hunk 1 above)
Comment (zvonimirfras) [reply src/slider/slider.component.ts:196]: I had similar thoughts.

File: src/slider/slider.component.ts [hunk 2]
```diff
@@ -72,6 +72,14 @@ import { EventService } from "carbon-components-angular/utils";
 						(mousedown)="onMouseDown($event)"
 						(keydown)="onKeyDown($event)">
 					</div>
+					<div
+						#thumb2
```
Comment (theiliad) [inline src/slider/slider.component.ts:76]: Just a heads up, the slider in the future should add more aria-* attributes to be properly accessible.

Comment (theiliad) [reply src/slider/slider.component.ts:76]: https://www.w3.org/TR/wai-aria-practices-1.1/examples/slider/slider-1.html

Comment (theiliad) [reply src/slider/slider.component.ts:76]: https://www.w3.org/TR/wai-aria-practices-1.1/examples/slider/slider-2.html

**Review by theiliad — CHANGES_REQUESTED** [review]
agreed wrt all the comments around value 1 & 2. The code can be a lot more flexible than that

File: src/slider/slider.component.ts [hunk 3]
```diff
@@ -147,38 +167,71 @@ export class Slider implements AfterViewInit, ControlValueAccessor {
...
-		if (this.input) {
-			this.input.value = v.toString();
+		if (this.inputs && this.inputs.length) {
+			this.inputs.forEach((input, index) => {
+				input.value = this._value[index].toString();
+			});
 		}
 
 		this.propagateChange(v);
 		this.valueChange.emit(v);
```
Comment (cal-smith) [inline src/slider/slider.component.ts:225]: Right now these will always emit an array

```suggestion
		const valueToEmit = this.isRange() ? v : v[0];
		this.propagateChange(valueToEmit);
		this.valueChange.emit(valueToEmit);
```

Comment (cal-smith) [reply src/slider/slider.component.ts:225]: Thankfully this is the only place we call those!

**Review by cal-smith — CHANGES_REQUESTED** [review]
This is actually kind of amazing ... It just works!

One small thing leftover though

File: src/slider/slider.component.ts (hunk 3 above)
Comment (zvonimirfras) [reply src/slider/slider.component.ts:225]: Ah yes, didn't come up in testing, good catch!

**Review by cal-smith — APPROVED** [review]

**Comment by stanislavgeorgiev** [comment]
Thanks for putting up the work on this item guys!

**Comment by cal-smith** [comment]
And thank you @stanislavgeorgiev for getting this conversation started!

**Pull request merged**